pkg/logs/input/container/scanner.go

```diff
@@ -171,11 +174,21 @@ func (s *Scanner) listContainers() []types.Container {
 
 // sourceShouldMonitorContainer returns whether a container matches a log source configuration.
 // Both image and label may be used:
-// - If the source defines an image, the container must match it exactly.
+// - If the source defines an image, the image must match with container with format "Y/image@sha256:X" where Y and @sha256:X are optional.
 // - If the source defines one or several labels, at least one of them must match the labels of the container.
 func (s *Scanner) sourceShouldMonitorContainer(source *config.LogSource, container types.Container) bool {
-	if source.Config.Image != "" && container.Image != source.Config.Image {
-		return false
+	if source.Config.Image != "" {
+		image := container.Image
+		if strings.Contains(image, digestPrefix) {
+			// Trim digest with format @256:X
+			splitted := strings.SplitN(image, digestPrefix, 2)
+			image = splitted[0]
+		}
+		// Expect prefix to be a repository or null
```

Comment:
`// Expect prefix to be a repository or null`
->
`// Expect prefix to end by '/'`?